compiler/rustc_const_eval/src/const_eval/valtrees.rs

```diff
@@ -323,7 +323,9 @@ pub fn valtree_to_const_value<'tcx>(
 
             valtree_into_mplace(&mut ecx, &place, valtree);
             dump_place(&ecx, &place);
-            intern_const_alloc_recursive(&mut ecx, InternKind::Constant, &place).unwrap();
+            assert!(
+                intern_const_alloc_recursive(&mut ecx, InternKind::Constant, &place).no_errors()
+            );
```

Comment:
I'm not a fan of having `assert!` contain relevant side-effects. Can you add `fn assert_no_errors`  to the `InternResult` type instead of `fn no_errors`?

Reply:
I'll just make it unwrappable again